src/Models/AzureComponentConfig.ts

```diff
@@ -40,38 +40,60 @@ export class AzureConfigFileHandler {
   private projectRootPath: string;
   private configFilePath: string;
 
+  private exists:((localPath: string) => Promise<boolean>) | undefined;
+  private writeFile: ((filePath: string, data: string | Buffer) => Promise<void>) | undefined;
+  private mkdirRecursively: ((dirPath: string) => Promise<void>) | undefined;
+
   constructor(projectRoot: string) {
     this.projectRootPath = projectRoot;
     this.configFilePath = path.join(
         this.projectRootPath, AzureComponentsStorage.folderName,
         AzureComponentsStorage.fileName);
   }
 
-  createIfNotExists() {
+
+  async createIfNotExistsInWorkspace() {

```

Comment:
combine those 2 API, use a variable to indicate whether this is local/remote in constructor.